src/channel/mpmc.rs

```diff
@@ -754,15 +775,17 @@ mod if_alloc {
             where
                 T: Send,
             {
-                generic_channel::<parking_lot::RawMutex, T>(capacity)
+                generic_channel::<parking_lot::RawMutex, T, HeapRingBuf<T>>(
```

Comment:
Actually wondering whether the default should now be  `GrowingRingBuf`. That defers some memory allocation costs, and is what Go is doing. Downside is some reallocation costs and worse performance on the first transmissions, and a less deterministic overall memory consumption (it could look fine in some small tests, but get bad in a long running program).

Going from `HeapRingBuf` to `GrowingRingBuf` is not a breaking change - but the other way around could be. So going back would not be as easy.

Reply:
> Actually wondering whether the default should now be GrowingRingBuf

Yeah that's kinda what I was thinking, I forgot to mention it. Thats what tokio is doing with their from what I recall. Really unsure though.

In my case, using this ring buf would be for unbounded channels where I don't care how much space they allocate because they are control flowed by another external mechanism.

I think that the main upside of the `HeapRingBuf` is that we allocate a continuous memory block, which should improve performance. I personally would always use a `HeapRingBuf` when I want a fixed size channel.

Reply:
`HeapRingBuf` also uses a `VecDeque`. And `VecDeque` uses a continous memory block :-)

 The diffference is really only about allocating the whole space upfront or reallocating if necessary. Maybe if we make that change it makes sense to rename also `HeapRingBuf` since both are now on the heap. `StaticHeapRingBuf` and `GrowingHeapRingBuf`? Unfortunately those are all pretty much on the long side.

Reply:
> HeapRingBuf also uses a VecDeque. And VecDeque uses a continous memory block :-)

Right, but I didn't know it reallocated an entire block when growing.

> Unfortunately those are all pretty much on the long side.

Maybe `StaticHeap` and `GrowingHeap` would be better.

Reply:
I like having `Buf` in the name - since the thing that we are referring to clearly isn't the `Heap`. It just lives on the heap. But I can live without `Ring`. So maybe `StaticHeapBuf` and `GrowingHeapBuf`. I think then the `Ring` should also be removed from `Array` variant - which then gets `ArrayBuf` or `StaticArrayBuf`. Instead of the `Static` we could also go for `Fixed`. 1 letter less

Reply:
I think `ArrayBuf` works since an array can't grow. I think I prefer `Fixed` over `Static` since I would assumed something named static to have a `'static` lifetime.